android/chromium_src/components/url_formatter/url_fixer.cc

```diff
@@ -0,0 +1,33 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this file,
+ * You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+#define IsEquivalentScheme IsEquivalentScheme_ChromiumImpl
+#include "../../../../../../components/url_formatter/url_fixer.cc"
+#undef IsEquivalentScheme
+
+namespace url_formatter {
+
+namespace {
+
+const char kBraveUIScheme[] = "brave";
+
+bool IsEquivalentScheme_BraveImpl(const std::string& scheme1,
+                                  const std::string& scheme2) {
+  return (scheme1 == kBraveUIScheme && scheme2 == kChromeUIScheme) ||
+         (scheme1 == kChromeUIScheme && scheme2 == kBraveUIScheme) ||
+         (scheme1 == url::kAboutScheme && scheme2 == kBraveUIScheme) ||
+         (scheme1 == kBraveUIScheme && scheme2 == url::kAboutScheme);
```

Comment:
I think maybe you need to compare about and chrome schemes here?

Reply:
Yes, I need this also, as I need to compare for equality all the pairs from `<brave, chrome, about>`:
```
brave<->chrome 
brave<->about
chrome<->about
```
But `chrome<->about` comparison is already done in IsEquivalentScheme_ChromiumImpl , https://cs.chromium.org/chromium/src/components/url_formatter/url_fixer.cc?type=cs&q=IsEquivalentScheme&g=0&l=683 
```
bool IsEquivalentScheme(const std::string& scheme1,
                        const std::string& scheme2) {
  return scheme1 == scheme2 ||
         (scheme1 == url::kAboutScheme && scheme2 == kChromeUIScheme) ||
         (scheme1 == kChromeUIScheme && scheme2 == url::kAboutScheme);
}
``` 
So I used compare `chrome<->about` by invoking `IsEquivalentScheme_ChromiumImp` .